lib/httplib/httpheaders.go

```diff
@@ -118,19 +132,31 @@ func SetDefaultSecurityHeaders(h http.Header) {
 	h.Set("Strict-Transport-Security", "max-age=31536000; includeSubDomains")
 }
 
-// SetIndexContentSecurityPolicy sets the Content-Security-Policy header for main index.html page
-func SetIndexContentSecurityPolicy(h http.Header, cfg proto.Features) {
-	cspMaps := []map[string]string{
-		defaultContentSecurityPolicy,
-		defaultFontSrc,
-		{"connect-src": "'self' wss:"},
-	}
+func getIndexContentSecurityPolicy(withStripe, withWasm bool) cspMap {
+	cspMaps := []cspMap{defaultContentSecurityPolicy, defaultFontSrc, defaultConnectSrc}
 
-	if cfg.GetCloud() && cfg.GetIsUsageBased() {
+	if withStripe {
 		cspMaps = append(cspMaps, stripeSecurityPolicy)
 	}
 
-	cspString := getContentSecurityPolicyString(cspMaps...)
+	if withWasm {
+		cspMaps = append(cspMaps, wasmSecurityPolicy)
+	}
+
+	return combineCSPMaps(cspMaps...)
+}
+
+// desktopSessionRe is a regex that matches /web/cluster/:clusterId/desktops/:desktopName/:username
+// which is a route to a desktop session that uses WASM.
```

Comment:
Technically this is false as of this commit (we're not yet using wasm), however it will be true once the `ironrdp` feature branch gets merged. Technically adding it now slightly increases our attack surface for no benefit until `ironrdp` is merged. I figure that since we're comfortable with this in eventuality, it's not such a security risk as to be worth worrying about.